### ISSUE TITLE
Is there any way to capture Sigint while blocked by stdin.readLineSync()?

### ISSUE THREAD

**Issue by busslina** [dart-lang/sdk#50943]
`ProcessSignal.sigint.watch().listen` works fine. But while reading stdin (`stdin.readLineSync()`) and you type Ctrl + c, you must wait the reading the whole line to get your signal callback executed.

My case is that I have an interactive menu, so I have 1..n options and I'm waiting the user to type the desired option index. I would like to capture Ctrl + c (sigint) signal at any time to exit properly.

Thanks

**Comment by rmacnak-google** [comment]
No, an isolate cannot do anything at all until a blocking call like `readSync` or `sleep` returns. If you want to wait for the next input from either stdin or a signal, use stdin as a Stream instead.

**Comment by busslina** [comment]
> 

Thank you,

Currently I'm using block method readByteSync and everything works fine (except this issue).

I tested the way you propose me, as Stream, and the problem I was getting there was related with restoring stdin echo mode and line mode after reading stream. It was something like OS error with file descriptor. I readed some post in which someone said it was normal that file descriptor does not exists after reading the stream. I disagree with that. My program was not finished, so fd must be available after reading the stream (await for and line mode to false). So that is the reason I switched to the sync way (byte after byte), all works fine except capturing signals.

I am missing something in the async way?
